cmd/agent/subcommands/run/command.go

```diff
@@ -438,9 +437,10 @@ func startAgent(
 			log.Errorf("Failed to start config management service: %s", err)
 		}
 
-		if err := rcclient.Listen("core-agent", []data.Product{data.ProductAgentTask, data.ProductAgentConfig}); err != nil {
+		if err := rcclient.Start("core-agent"); err != nil {
 			pkglog.Errorf("Failed to start the RC client component: %s", err)
 		}
+		rcclient.SubscribeAgentTask()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Do you really want to call that function if there was an error in `Start` ?
If so maybe adding a comment would be nice because it seems unintuitive